flipperdevices/flipperzero-firmware
# [FL-2728] Fix Null Pointer Dereference errors in IR

# What's new

- Fixed Null pointer dereference errors in Infrared App
- Cleaned up code

# Verification 

1. Go to Infrared -> Learn New Remote
2. Add several new buttons by repeatedly pressing the `+` button. Use a RC5(X) remote.
3. No crashes should occur.
4. Use the IR App as usual. No crashes should occur.

# Checklist (For Reviewer)
